crates/aggregator/src/committee_finalizer.rs

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+//
+// This file is provided WITHOUT ANY WARRANTY;
+// without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE.
+
+use actix::prelude::*;
+use e3_events::{CommitteeRequested, EnclaveEvent, EventBus, Shutdown, Subscribe};
+use e3_evm::FinalizeCommittee;
+use std::collections::HashMap;
+use std::time::Duration;
+use tracing::{error, info};
+
+/// CommitteeFinalizer is an actor that listens to CommitteeRequested events and calls
+/// finalizeCommittee on the registry after the submission deadline has passed.
+pub struct CommitteeFinalizer {
+    #[allow(dead_code)]
+    bus: Addr<EventBus<EnclaveEvent>>,
+    registry_writer: Recipient<FinalizeCommittee>,
```

Comment:
Whilst using Recipient allows for polymorphic testing we are still passing in a direct reference. Instead of passing a direct reference like this I would suggest we consider dispatching a `ComitteeFinalizeRequested` event on the bus so that core and shell know nothing about each other and things are totally separated. We then don't need to do anything fancy on construction and can just pass the bus in